wc1994.py

```diff
@@ -0,0 +1,160 @@
+        if rake is None:
+            # their "All" case
+            return 0.24
+        elif (-45 <= rake <= 45) or (rake >= 135) or (rake <= -135):
+            # strike slip
+            return 0.22
+        elif rake > 0:
+            # thrust/reverse
+            return 0.26
+        else:
+            # normal
+            return 0.22
+
+    def get_std_dev_mag(self, rake):
+        """
+        Standard deviation on the magnitude for the WC1994 area relation.
+        """
+        assert rake is None or -180 <= rake <= 180
+        if rake is None:
+            # their "All" case
+            return 0.24
+        elif (-45 <= rake <= 45) or (rake >= 135) or (rake <= -135):
+            # strike slip
+            return 0.23
+        elif rake > 0:
+            # thrust/reverse
+            return 0.25
+        else:
+            # normal
+            return 0.25
+
+    def get_median_mag(self, area, rake):
+        """
+        Return magnitude (Mw) given the area and rake.
+
+        Setting the rake to ``None`` causes their "All" rupture-types
+        to be applied.
+
+        :param area:
+            Area in square km.
+        :param rake:
+            Rake angle (the rupture propagation direction) in degrees,
+            from -180 to 180.
+        """
+        assert rake is None or -180 <= rake <= 180
+        if rake is None:
+            # their "All" case
+            return 4.07 + 0.98 * log10(area)
+        elif (-45 <= rake <= 45) or (rake > 135) or (rake < -135):
+            # strike slip
+            return 3.98 + 1.02 * log10(area)
+        elif rake > 0:
+            # thrust/reverse
+            return 4.33 + 0.90 * log10(area)
+        else:
+            # normal
+            return 3.93 + 1.02 * log10(area)
+
+"""
+Local modification of WC1994 to mimic behaviour of GSCFRISK code for the 
+Queen Charlotte Strike-Slip (QCSS) fault based on rupture length for the
+2015 Seismic Hazard Model of Canada as documented in Adams, J., S. Halchuk, 
+T. Allen, and G. Rogers (2015). Canada's 5th Generation seismic hazard model, 
+as prepared for the 2015 National Building Code of Canada, 11th Canadian 
+Conference on Earthquake Engineering, Victoria, Canada, Paper 93775.
+"""
+class WC1994_QCSS(BaseMSRSigma):
+    """
+    Implements magnitude-length scaling relationship for strike-slip faults
+    
+    Coefficents taken from Table 2A (P990) of Wells, D. L., and K. J. 
+    Coppersmith (1994). New empirical relationships among magnitude, 
+    rupture length, rupture width, rupture area, and surface displacement, 
+    Bull. Seism. Soc. Am. 84, 974-1002.
+    
+    :param length:
+        length for strike slip faults (km)
+    :param seis_wid:
+        hirdwired seismogenic width of the QCSS (20 km)
+    """
+    def get_median_area(self, mag, rake):
+        """
+        The values are a function of magnitude.
+        """
+        # strike slip
+        length = 10.0 ** (-2.57 + 0.67 * mag)
```

Comment:
Shouldn't this be `10.0 ** (-2.57 + 0.62 * mag)` - based on Table 2A of Wells & Coppersmith (1994)